functions.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Mm Custom Add-ons General Functionality.
+ *
+ * @since 1.2.0
+ *
+ * @package mm-add-ons
+ */
+
+add_filter( 'mm_shortcode_custom_classes', 'mm_shortcode_custom_classes', 10, 3 );
+/**
+ * Add custom shortcode classes.
+ *
+ * @since   1.2.0
+ *
+ * @param   string  $classes  Initial classes.
+ * @param   string  $tag      Shortcode tag.
+ * @param   array   $atts     Shortcoder atts.
+ *
+ * @return  string            Modified classes.
+ */
+function mm_shortcode_custom_classes( $classes, $tag, $atts ) {
+
+	/**
+	 * Add classes for custom shortcode attributes
+	 *
+	 * These classes are defined, and identifiable, with the following format:
+	 * mm_class_*
+	 */
+
+	// Set up classes array.
+	$class_array = explode( ' ', $classes );
+
+	// Loop through each att and add class as needed.
+	foreach ( $atts as $key => $value ) {
+
+		// Add class in the following format: $key-$class
+		// Exclude custom class att as this needs to be unprefixed.
+		if ( false !== strpos( $key, 'mm_class_' ) ) {
+			$class_array[] = $key . '-' . str_replace( ',', ' ', $value );
```

Comment:
I think I understand that by replacing commas with whitespace you're trying to support multiple classes coming in a single shortcode param, like classes="blue,large", but in this case wouldn't the key only get added to the first class? Like "key-blue large"?